subsys/bluetooth/services/rscs.c

```diff
@@ -0,0 +1,426 @@
+	case SC_CP_OP_CALIBRATION:
+		return rscs_inst.features.sensor_calib_proc;
+	case SC_CP_OP_UPDATE_LOC:
+	case SC_CP_OP_REQ_SUPP_LOC:
+		return rscs_inst.features.multi_sensor_loc;
+	default:
+		return false;
+	}
+}
+
+struct write_sc_ctrl_point_req {
+	uint8_t op;
+	union {
+		uint32_t total_dist;
+		uint8_t location;
+	};
+} __packed;
+
+static bool ctrl_point_param_validate(const struct write_sc_ctrl_point_req *req, uint16_t len)
+{
+	switch (req->op) {
+	case SC_CP_OP_SET_CUMULATIVE:
+		return len == sizeof(req->op) + sizeof(req->total_dist);
+	case SC_CP_OP_UPDATE_LOC:
+		return len == sizeof(req->op) + sizeof(req->location);
+	case SC_CP_OP_CALIBRATION:
+	case SC_CP_OP_REQ_SUPP_LOC:
+		return len == sizeof(req->op);
+	default:
+		return false;
+	}
+}
+
+static ssize_t write_ctrl_point(struct bt_conn *conn, const struct bt_gatt_attr *attr,
+				const void *buf, uint16_t len, uint16_t offset, uint8_t flags)
+{
+	const struct write_sc_ctrl_point_req *req = buf;
+	uint8_t status;
+
+	if (!rscs_inst.ctrl_pt_indicate_enabled) {
+		return BT_GATT_ERR(RSC_ERR_CCC_CONFIG);
+	}
+
+	if (!len) {
+		return BT_GATT_ERR(BT_ATT_ERR_INVALID_ATTRIBUTE_LEN);
+	}
+
+	if (!ctrl_point_op_code_validate(req->op)) {
+		LOG_ERR("Operation code not supported %#x", req->op);
+		ctrl_point_ind(conn, req->op, SC_CP_RSP_OP_NOT_SUPP, NULL, 0);
+		return len;
+	}
+
+	if (!ctrl_point_param_validate(req, len)) {
+		LOG_ERR("Invalid parameter for op code %#x", req->op);
+		ctrl_point_ind(conn, req->op, SC_CP_RSP_INVAL_PARAM, NULL, 0);
+		return len;
+	}
+
+	switch (req->op) {
+	case SC_CP_OP_SET_CUMULATIVE:
+		if (rscs_inst.cp_cb && rscs_inst.cp_cb->set_cumulative) {
+			if (!rscs_inst.cp_cb->set_cumulative(sys_le32_to_cpu(req->total_dist))) {
+				status = SC_CP_RSP_SUCCESS;
+				break;
+			}
+		}
+		status = SC_CP_RSP_FAILED;
+		break;
+	case SC_CP_OP_CALIBRATION:
+		if (rscs_inst.cp_cb && rscs_inst.cp_cb->calibrarion) {
+			if (!rscs_inst.cp_cb->calibrarion()) {
+				status = SC_CP_RSP_SUCCESS;
+				break;
+			}
+		}
+		status = SC_CP_RSP_FAILED;
+		break;
+	case SC_CP_OP_UPDATE_LOC:
+		status = SC_CP_RSP_INVAL_PARAM;
+		for (size_t i = 0; i < rscs_inst.num_location; i++) {
+			if (rscs_inst.supported_locations[i] == req->location) {
+				rscs_inst.sensor_loc = req->location;
+				status = SC_CP_RSP_SUCCESS;
+				break;
+			}
+		}
+
+		if (status == SC_CP_RSP_SUCCESS && rscs_inst.cp_cb && rscs_inst.cp_cb->update_loc) {
+			rscs_inst.cp_cb->update_loc(rscs_inst.sensor_loc);
+		}
+		break;
+	case SC_CP_OP_REQ_SUPP_LOC:
+		ctrl_point_ind(conn, req->op, SC_CP_RSP_SUCCESS, &rscs_inst.supported_locations,
+			       rscs_inst.num_location);
+		return len;
+	default:
+		status = SC_CP_RSP_OP_NOT_SUPP;
+		break;
+	}
+
+	ctrl_point_ind(conn, req->op, status, NULL, 0);
+	return len;
+}
+
+BT_GATT_SERVICE_DEFINE(rsc_svc,
+	BT_GATT_PRIMARY_SERVICE(BT_UUID_RSCS),
+	BT_GATT_CHARACTERISTIC(BT_UUID_RSC_MEASUREMENT, BT_GATT_CHRC_NOTIFY,
+			       0x00, NULL, NULL, NULL),
```

Comment:
instead of 0x00:
```suggestion
			       BT_GATT_PERM_NONE, NULL, NULL, NULL),
```

but are you sure that thid char shouldn't have at least read permission ?